core/src/main/codegen/templates/Parser.jj

```diff
@@ -1559,6 +1559,11 @@ SqlNode SelectItem() :
     SqlIdentifier id;
 }
 {
+    LOOKAHEAD(7)
+    e = StarIdentifier() {
+        return e;
+    }
+    |
     e = SelectExpression()
     [
         [ <AS> ]
```

Comment:
instead of introducing a new identifier/etc you should be able to make the check here - and raise an exception if whether `e` is a star identifier